sys/arm64/arm64/locore.S

```diff
@@ -109,7 +193,48 @@ _start:
 	mrs x2, cctlr_el0
 	orr x2, x2, #(CCTLR_PERMVCT_MASK | CCTLR_SBL_MASK)
 	msr cctlr_el0, x2
+
+#ifdef __CHERI_PURE_CAPABILITY__
+	/*
+	 * Enable capablity sealing for branch and link at EL1
+	 * Use PCC/DDC address interpretation.
+	 * Use DDC as base for adrdp.
+	 */
+	mrs	x2, cctlr_el1
+	bic	x2, x2, #(CCTLR_PCCBO_MASK | CCTLR_DDCBO_MASK | CCTLR_ADRDPB_MASK)
+	orr	x2, x2, #(CCTLR_SBL_MASK)
+	msr	cctlr_el1, x2
+
+	/* We assume that we enter here in a64 mode. */
+	bx	#4
+	.arch_extension c64
 #endif
+#endif
+.endmacro
+
+	.text
+	.globl _start
+_start:
+#if defined(LINUX_BOOT_ABI)
+	/* U-boot image header */
+	b	1f			/* code 0 */
+	.long	0			/* code 1 */
+	.quad	UBOOT_IMAGE_OFFSET	/* Image offset in 2 MiB page, LE */
+	.quad	UBOOT_IMAGE_SIZE	/* Image size, LE */
+	.quad	UBOOT_IMAGE_FLAGS	/* Flags for kernel. LE */
+	.quad 	0			/* Reserved */
+	.quad 	0			/* Reserved */
+	.quad 	0			/* Reserved */
+	.long 	0x644d5241		/* Magic  "ARM\x64", LE */
+	.long 	0			/* Reserved for PE COFF offset*/
+1:
+#endif /* defined(LINUX_BOOT_ABI) */
+
+	/* Drop to EL1 */
+	bl	drop_to_el1
```

Comment:
Worth highlighting this is an A64 `bl`, i.e. it's linking an integer not a capability